### PR TITLE
Upgrade Kafka client-libraries

### PR DESCRIPTION
**Pull request by cstub**
* Upgrade Dependencies:
  - Upgrade kafka-client to version `0.11.0.0`
  - Upgrade akka-stream-kafka to version `0.17`
  - Upgrade akka to version `2.4.20`

* Remove `withProducerProvider` option from `TransactionalEventProducer.Settings`:
  Remove the `Producer`-provider from settings of the `TransactionalEventProducer` as this option is no longer necessary due to the default `Producer` dropping outstanding messages in case of failures since version `0.17` of akka-stream-kafka.

* Migrate JUnit tests to ScalaTest:
  Convert all JUnit tests to ScalaTest to improve readability and drop dependencies on JUnit and kafka-unit.

### CODE REVIEW

File: src/main/scala/com/rbmhtechnology/calliope/javadsl/TransactionalEventProducer.scala [hunk 1]
```diff
@@ -114,9 +114,6 @@ object TransactionalEventProducer {
     def withProducerProperty(key: String, value: String): Settings[A] =
       copy(delegate.withProducerProperty(key, value))
 
-    def withProducerProvider(producerProvider: ProducerProvider[A]): Settings[A] =
```
Comment (volkerstampa) [inline src/main/scala/com/rbmhtechnology/calliope/javadsl/TransactionalEventProducer.scala:117]: I guess this means we can also delete the type aliases for `ProducerProvider` in java- and scaladsl

File: src/test/scala/com/rbmhtechnology/calliope/javadsl/DeduplicationMapSpec.scala [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+          val map = EmptyDeduplicationMap
+            .append(Entry(Topic1, 0, "agg1", "p-0"))
+            .append(Entry(Topic1, 0, "agg1", "p-1"))
+            .append(Entry(Topic1, 0, "agg2", "p-2"))
+            .append(Entry(Topic1, 0, "agg2", "p-3"))
+            .append(Entry(Topic1, 0, "agg3", "p-4"))
+            .append(Entry(Topic1, 0, "agg3", "p-5"))
+
+          map.size must be(3)
+          map.asScala.toList must contain inOrderOnly(
```
Comment (volkerstampa) [inline src/test/scala/com/rbmhtechnology/calliope/javadsl/DeduplicationMapSpec.scala]: Cant we just say `mustBe List(...)` to get rid of the separate size-check above? (here and below in the other tests)

**Review by volkerstampa — APPROVED** [review]
LGTM. Just minor comments.

File: src/test/scala/com/rbmhtechnology/calliope/javadsl/DeduplicationMapSpec.scala (hunk 2 above)
Comment (cstub) [reply src/test/scala/com/rbmhtechnology/calliope/javadsl/DeduplicationMapSpec.scala]: the size-check is performed to check if the resulting size is calculated correctly in the `DeduplicationMap`

Comment (cstub) [reply src/test/scala/com/rbmhtechnology/calliope/javadsl/DeduplicationMapSpec.scala]: I will add a separate test for size-calculation though and remove the size-check in all other tests

**Pull request merged**